GVFS/GVFS/RepairJobs/GitRefsHeadsRepairJob.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using GVFS.Common;
+using GVFS.Common.Tracing;
+
+namespace GVFS.RepairJobs
+{
+    public class GitRefsHeadsRepairJob : GitRefsRepairJob
+    {
+        public GitRefsHeadsRepairJob(ITracer tracer, TextWriter output, GVFSEnlistment enlistment)
+            : base(tracer, output, enlistment)
+        {
+        }
+
+        public override string Name
+        {
+            get { return @".git\refs\heads\**\*"; }
+        }
+
+        protected override IEnumerable<string> GetRefs()
+        {
+            string refsHeadsPath = Path.Combine(this.Enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Refs.Heads.RootFolder);
+            string dotGitPath = Path.Combine(this.Enlistment.WorkingDirectoryRoot, GVFSConstants.DotGit.Root) + Path.DirectorySeparatorChar;
+
+            return Directory.EnumerateFiles(refsHeadsPath, "*", SearchOption.AllDirectories)

```

Comment:
I think a better way to write this code would be to use `DirectoryInfo` to enumerate the children, so that you can keep everything in relative-path-space, and not have to convert from relative to full back to relative paths. As you've seen, paths are hard, and I can't say with any confidence that the current `GetRelativePath` implementation is correct, or even what it might be missing. We should avoid manipulating them to whatever extent possible.